pkg/serializer/convertor.go

```diff
@@ -177,9 +177,39 @@ func (c *objectConvertor) ConvertToVersion(in runtime.Object, _ runtime.GroupVer
 	if ok {
```

Comment:
If we requested to encode as the convertible gvk, this will in error encode as the hub